src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1134,7 +1163,27 @@ bool entitiesQuery
 
         /* Attributes part */
 
-        std::vector<BSONElement> queryAttrV = r.getField(ENT_ATTRS).Array();
+        std::vector<BSONElement> queryAttrV;
+
+        //
+        // This try/catch should not be necessary as all document in the entities collection have an attrs array
+        // from creation time. However, it adds an extra protection, just in case.
+        // Somebody *could* manipulate the mongo database and if so, the broker would crash here.
+        // Better to be on the safe side ...
+        //
+        try
+        {
+          queryAttrV = r.getField(ENT_ATTRS).Array();
+        }
+        catch (...)
+        {
+          LM_E(("Database Error (no attrs array in document of entities collection)"));
+          cer->statusCode.fill(SccReceiverInternalError);
```

Comment:
Maybe we should use some details string to explain what had happened, e.g. "attrs field missing in entity document".